nist/int.go

```diff
@@ -314,21 +313,17 @@ func (i *Int) UnmarshalBinary(buf []byte) error {
 	if len(buf) != i.MarshalSize() {
 		return errors.New("Int.Decode: wrong size buffer")
 	}
-	i.V.SetBytes(buf)
+	if i.endianness == binary.LittleEndian {
+		i.SetLittleEndian(buf)
+	} else {
+		i.V.SetBytes(buf)
```

Comment:
Well, if `MarshalBinary` returns a slice in LittleEndian format, then `UnmarshalBinary` on the same `Int` should expect a slice in LittleEndian format ... I thought it was the whole point of the thing,  to be able to use  `Secret` which `Marshal/Unmarshal` in LittleEndian format... ?